synapse/events/validator.py

```diff
@@ -185,3 +237,8 @@ def _ensure_strings(self, d, keys):
     def _ensure_state_event(self, event):
         if not event.is_state():
             raise SynapseError(400, "'%s' must be state events" % (event.type,))
+
+    def _is_object(self, checker, instance):
+        if isinstance(instance, (dict, frozendict.frozendict)):
```

Comment:
we normally write this:

```suggestion
        if isinstance(instance, collections.abc.Mapping):
```